src/modules/expenses/expense.routes.ts

```diff
@@ -0,0 +1,86 @@
+import { createRoute } from "@hono/zod-openapi";
+import { z } from "zod";
+
+import jsonContentRequired from "@/common/helpers/json-content-required.helper";
+import { jsonContent } from "@/common/helpers/json-content.helper";
+import {
+  authMiddleware,
+  requireAuth,
+} from "@/common/middlewares/auth.middleware";
+import createErrorSchema from "@/common/schema/create-error.schema";
+import * as HTTPStatusCodes from "@/common/utils/http-status-codes.util";
+import {
+  insertExpenseSchema,
+  selectExpenseSchema,
+} from "@/db/schemas/expense.model";
+
+const tags = ["Expenses"];
+
+export const createExpenseRoute = createRoute({
+  tags,
+  method: "post",
+  path: "/expenses",
+  middleware: [
+    authMiddleware(),
+    requireAuth(),
+  ] as const,
+  request: {
+    body: jsonContentRequired(
+      insertExpenseSchema
+        .omit({
+          id: true,
+          createdAt: true,
+          updatedAt: true,
+          creatorId: true,
+        })
+        .partial({
+          payerId: true,
+        }),
+      "Expense creation details",
+    ),
+  },
+  responses: {
+    [HTTPStatusCodes.CREATED]: jsonContent(
+      z.object({
+        success: z.boolean().default(true),
+        message: z.string(),
+        data: selectExpenseSchema,
+      }),
+      "Expense created successfully",
+    ),
+    [HTTPStatusCodes.BAD_REQUEST]: jsonContent(
+      z.object({
+        success: z.boolean().default(false),
+        message: z.string(),
+      }),
+      "The validation error(s)",
```

Comment:
These common messages going forward should come from the exported constants.

Reply:
Done.